rllib/algorithms/algorithm.py

```diff
@@ -856,13 +856,12 @@ def setup(self, config: AlgorithmConfig) -> None:
                     # Provide the actor handles for the learners for module
                     # updating during preprocessing.
                     self.offline_data.learner_handles = self.learner_group._workers
-                    # Provide the module_spec. Note, in the remote case this is needed
-                    # because the learner module cannot be copied, but must be built.
-                    self.offline_data.module_spec = module_spec
                 # Otherwise we can simply pass in the local learner.
                 else:
                     self.offline_data.learner_handles = [self.learner_group._learner]
-
+                # Provide the module_spec. Note, in the remote case this is needed
```

Comment:
nit: Can we add a TODO here to replace these set-some-object's-attributes-directly? We should find some solution for this in the future, an API, or setting these in the OfflineData constructor?